installer/scripts/service_control

```diff
@@ -191,31 +191,34 @@ find_systemd_dir()
 
 enable_omsagent_service()
 {
-    echo "Configuring OMS agent service ..."
-    if pidof systemd 1> /dev/null 2> /dev/null; then
-        # systemd
-        local systemd_dir=$(find_systemd_dir)
-        cp /etc/opt/microsoft/omsagent/sysconf/omsagent.systemd ${systemd_dir}/omsagent.service
-        /bin/systemctl daemon-reload
-        /bin/systemctl -q enable omsagent
-    else
-        cp /etc/opt/microsoft/omsagent/sysconf/omsagent.ulinux /etc/init.d/omsagent
-
-        if [ -x /usr/sbin/update-rc.d ]; then
-            update-rc.d omsagent defaults > /dev/null
-        elif [ -x /usr/lib/lsb/install_initd ]; then
-            /usr/lib/lsb/install_initd /etc/init.d/omsagent
-        elif [ -x /sbin/chkconfig ]; then
-            chkconfig --add omsagent > /dev/null
+    if [ ! -f /etc/opt/microsoft/omsagent/conf/.service_registered ] && [ -f /etc/opt/microsoft/omsagent/conf/omsadmin.conf ]; then
+        echo "Configuring OMS agent service ..."
+        if pidof systemd 1> /dev/null 2> /dev/null; then
+            # systemd
+            local systemd_dir=$(find_systemd_dir)
+            cp /etc/opt/microsoft/omsagent/sysconf/omsagent.systemd ${systemd_dir}/omsagent.service
+            /bin/systemctl daemon-reload
+            /bin/systemctl -q enable omsagent
         else
-            echo "Could not find a service controller to configure the OMS Agent Service."
-            exit 1
+            cp /etc/opt/microsoft/omsagent/sysconf/omsagent.ulinux /etc/init.d/omsagent
+
+            if [ -x /usr/sbin/update-rc.d ]; then
+                update-rc.d omsagent defaults > /dev/null
+            elif [ -x /usr/lib/lsb/install_initd ]; then
+                /usr/lib/lsb/install_initd /etc/init.d/omsagent
+            elif [ -x /sbin/chkconfig ]; then
+                chkconfig --add omsagent > /dev/null
+            else
+                echo "Could not find a service controller to configure the OMS Agent Service."
+                exit 1
+            fi
         fi
+        touch  /etc/opt/microsoft/omsagent/conf/.service_registered
```

Comment:
Since this script creates the marker file, this script should delete the marker file as per our IM conversation. Please take care of that.